CONTRIBUTING.md

```diff
@@ -26,7 +26,54 @@ including the linter (which checks for coding style and warnings, see [Style gui
 These checks are ran automatically using [GitHub Actions](#github-actions). You also need at least
 one approval from a maintainer - after submitting a PR, you can request a review from the top right
 Reviewers menu on the Pull Request page.
+### Branches
+Please use the following structure for creating new development branches:
+```
+[username]/[snake_case_branch_name]
+```
+This will make branches easier to find and search for.<br>
+Examples:
+```
+razvanra2/add_authoring_policy
+luigi/dev_eng_improv
+auditore/fix_md_typo
+torvalds/android_speedups
+```
+### Merging
+When developing a new feature or working on a bug, your pull request will end up containing fixup commits (commits that change the same line of code repeatedly) or too fine-grained commits. An issue that can arise from this is that the main branch history will become poluted with unnecessary commits. To avoid it, we implement and enforce a squash policy.
+All commits that are merged intro the main development branch have to be squashed ahead of the merge.
+You can do so by pressing "squash and merge", or, alternetively, following the generic local squash routine outlined bellow:
+```
+git checkout your_branch_name
+git rebase -i HEAD~n
+# n is normally the number of commits in the pull request.
+# Set commits (except the one in the first line) from 'pick' to 'squash', save and quit.
+# On the next screen, edit/refine commit messages.
+# Save and quit.
+git push -f # (force push to GitHub)
+```
+Please update the resulting commit message, if needed. It should read as a coherent message. In most cases, this means not just listing the interim commits.
+
+Please refrain from creating several pull requests for the same change. Use the pull request that is already open (or was created earlier) to amend changes. This preserves the discussion and review that happened earlier for the respective change set.
+Similarly, please create one PR per development item, instead of bundling multiple fixes and improvments in a singule PR.
+The length of time required for peer review is unpredictable and will vary from pull request to pull request.
 
+Please also prefix the issue number between square brackets in the title of a PR. This will, in the future, make it simpler to track how various decisions where taken during the development process.<br>
```

Comment:
Not sure how I feel about this. I would prefer just mentioning "This closes #n" or "This fixes #n" in the PR description, since that way the issue is automatically linked to the PR (and closed when the PR is merged) and you can see it when you open the PR.
I would suggest we use prefixes like this instead: https://forum.grin.mw/t/github-pr-pull-request-title-prefix-rule-optional/842

Reply:
We could add this to the branch names too. Like `razvanra2/fix/add_event_bug`. Unless you think it's better to have `razvanra2/fix_add_event_bug`, not sure which version would be easier to maintain.

Reply:
This PR is actually a good example - sure, it has 60 in the title, but it's not actually linked to issue #60, as far as GitHub is concerned. Plus there's also the problem of - just the number in the title doesn't tell me whether the PR *closes* the issue or is just related to it.

Reply:
Works too, it's just my personal preference. Let's see what the others think

Reply:
old but gold [article ](https://chris.beams.io/posts/git-commit/) on this

Reply:
That's a classic and we should definitely add it to this documentation, but we still need to define guidelines for the PR title & description specifically.

Reply:
Alternatively - or maybe additionally - we could also enforce the usage of labels for PRs. We currently only use some for issues.

Reply:
About the name of branches, I would go with `username/branch`. Usually each person has 1 branch at any moment, in special cases they may have 2 or 3, so it should be pretty quick to find a branch if u search by username. I wouldn't complicate things fix enforcing other markers in the branch name,